pkg/generate/dockerfile/finder.go

```diff
@@ -42,26 +42,26 @@ func NewFinder() Finder {
 	return &finder{fsWalk: filepath.Walk}
 }
 
-// Find returns a list of of found Dockerfile(s) in the given directory
+// Find returns the relative paths of Dockerfiles in the given directory dir.
 func (f *finder) Find(dir string) ([]string, error) {
 	dockerfiles := []string{}
 	err := f.fsWalk(dir, func(path string, info os.FileInfo, err error) error {
-		if info == nil {
-			return nil
+		if err != nil {
+			return err
 		}
-		// Skip hidden directories
+		// Skip hidden directories.
 		if info.IsDir() && strings.HasPrefix(info.Name(), ".") {
 			return filepath.SkipDir
 		}
-
-		// Add files named Dockerfile
-		if info.Name() == "Dockerfile" && err == nil {
+		// Add relative path to Dockerfile.
+		if info.Mode().IsRegular() && info.Name() == "Dockerfile" {
```

Comment:
> do we definitely want to skip symlinked dockerfiles?

I have no strong opinions here. We definitely don't want to return directories in the list of Dockerfiles.

We can include symlinks explicitly with:
```go
if (info.Mode().IsRegular() || info.Mode()&os.ModeSymlink != 0) && info.Name() == "Dockerfile" {
```

@bparees WDYT?